### PR TITLE
feat: confirmation re-designs add basic page for types sign V1 signature request

### PR DESCRIPTION
**Pull request by jpuri**
## **Description**

Confirmation re-designs: add basic page for sign types data V1 signature requests.

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/3638

## **Manual testing steps**

1. Enable re-designs locally
2. Open test dapp
3. Submit signed type V1 request and check page

## **Screenshots/Recordings**
Screenshot 2024-11-19 at 5 32 09 PM

## **Pre-merge author checklist**

- [X] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've completed the PR template to the best of my ability
- [X] I’ve included tests if applicable
- [X] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [X] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/components/Views/confirmations/components/Confirm/NoChangeSimulation/NoChangeSimulation.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+
+import { strings } from '../../../../../../../locales/i18n';
+import { selectUseTransactionSimulations } from '../../../../../../selectors/preferencesController';
+import InfoSection from '../../UI/InfoRow/InfoSection';
+import InfoRow from '../../UI/InfoRow';
+
+const NoChangeSimulation = () => {
+  const useTransactionSimulations = useSelector(
```
Comment (OGPoyraz) [inline app/components/Views/confirmations/components/Confirm/NoChangeSimulation/NoChangeSimulation.tsx:10]: Can we use this selector one level above like we did it for the real `SimulationDetails` so we can keep the same convention? Later it's being a problem to check all these selectors-arguments in deep level functions like sending metrics.

File: app/components/Views/confirmations/components/Confirm/Info/TypedSignV1/Message.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+import useApprovalRequest from '../../../../hooks/useApprovalRequest';
+import SignatureMessageSection from '../../SignatureMessageSection';
+
+const Message = () => {
+  const { approvalRequest } = useApprovalRequest();
+
+  const message = useMemo(
+    () => JSON.stringify(approvalRequest?.requestData?.data, undefined, 4),
+    [approvalRequest?.requestData?.data],
```
Comment (OGPoyraz) [inline app/components/Views/confirmations/components/Confirm/Info/TypedSignV1/Message.tsx:12]: Is it worth assigning this into a const and re-use?

**Review by OGPoyraz — DISMISSED** [review]

File: app/components/Views/confirmations/components/Confirm/NoChangeSimulation/NoChangeSimulation.tsx (hunk 1 above)
Comment (jpuri) [reply app/components/Views/confirmations/components/Confirm/NoChangeSimulation/NoChangeSimulation.tsx:10]: Keeping it one level above will require code replication alone each usage.

File: app/components/Views/confirmations/components/Confirm/Info/TypedSignV1/Message.tsx (hunk 2 above)
Comment (jpuri) [reply app/components/Views/confirmations/components/Confirm/Info/TypedSignV1/Message.tsx:12]: This is actually temporary, I am working on following PR to create data tree for typed sign data.

File: app/components/Views/confirmations/components/Confirm/Info/TypedSignV1/Message.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  );
+
+  return (
+    <SignatureMessageSection
+      messageCollapsed={message}
+      messageExpanded={<Text>{message}</Text>}
```
[the remark this answers is not in the document]
Comment (jpuri) [reply app/components/Views/confirmations/components/Confirm/Info/TypedSignV1/Message.tsx:18]: yep same as above this is temporary, I removed those styled here as sonar was complaining about code duplication.

File: app/components/Views/confirmations/Confirm/Confirm.test.tsx
[the remark this answers is not in the document]
Comment (jpuri) [reply app/components/Views/confirmations/Confirm/Confirm.test.tsx:1]: yep this is possible, but let me check with designs for this, I will fix this in following PR

**Review by OGPoyraz — APPROVED** [review]

File: app/components/Views/confirmations/components/Confirm/NoChangeSimulation/NoChangeSimulation.tsx (hunk 1 above)
Comment (OGPoyraz) [reply app/components/Views/confirmations/components/Confirm/NoChangeSimulation/NoChangeSimulation.tsx:10]: Same thing applies for `SimulationDetails` but it never tries to render. Also my point here is to keep the same conventions everywhere.

**Review by digiwand — APPROVED** [review]

**Pull request merged**